Source/JavaScriptCore/wasm/WasmOMGIRGenerator.cpp

```diff
@@ -2867,8 +2907,10 @@ Value* OMGIRGenerator::emitAtomicCompareExchange(ExtAtomicOpType op, Type valueT
         truncatedValue = m_currentBlock->appendNew<B3::Value>(m_proc, B3::Trunc, Origin(), value);
     }
 
-    auto result = m_currentBlock->appendNew<AtomicValue>(m_proc, memoryKind(AtomicStrongCAS), origin(), accessWidth, truncatedExpected, truncatedValue, pointer);
-    return sanitizeAtomicResult(op, valueType, result);
+    auto* atomic = m_currentBlock->appendNew<AtomicValue>(m_proc, memoryKind(AtomicStrongCAS), origin(), accessWidth, truncatedExpected, truncatedValue, pointer);
+    m_heaps.decorateMemory(&m_heaps.WebAssemblyMemory, atomic);
+    m_heaps.decorateFencedAccess(&m_heaps.WebAssemblyMemory, atomic);
```

Comment:
When would you `m_heaps.decorateFencedAccess` without `m_heaps.decorateMemory`? Can we push the `decorateMemory` into the fenced decoration?

Reply:
Yeah, that's true. We should just do both for atomic values. Done.

Reply:
Ah, but it seems that we are doing this separation intentionally to make relaxed atomics possible (if fenced one is not attached, then it is relaxed). So for now, I'll leave it.